source/SH3/arc/vfile.cpp

```diff
@@ -31,5 +57,16 @@ std::size_t sh3_arc_vfile::ReadData(void* destination, std::size_t len, read_err
         e.set_error(read_result::PARTIAL_READ);
     }
 
-    std::memcpy(destination, buffer.data() + fpos, len);
+    // Prevent memcpy from reading past the vector
+    if(fpos + len > fsize)
```

Comment:
`fpos + len` may overflow the datatype. You could do `len = std::min(len, fsize - fpos);` instead though; fpos should never be larger than fsize (you could `assert()` that).